### PR TITLE
add invariantHolds and isApplicable test for FriendlyBallPlacement

### PR DESCRIPTION
**Pull request by jameschuang2002**
### Description

add invariantHolds and isApplicable test for FriendlyBallPlacement

### Resolved Issues
resolves #2276

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/play/ball_placement_play_test.cpp [hunk 1]
```diff
@@ -46,3 +46,23 @@ TEST_F(BallPlacementPlayTest, test_ball_placement)
             terminating_validation_functions, non_terminating_validation_functions,
             Duration::fromSeconds(10));
 }
+TEST(FriendlyBallPlacementInvariantAndIsApplicableTest, test_invariant_and_is_applicable)
```
Comment (PhilD71) [inline src/software/ai/hl/stp/play/ball_placement_play_test.cpp:49]: Name is a bit of a mouthful, but not your fault, just the situation.

**Review by PhilD71 — APPROVED** [review]
Looks good to me!

**Review by akhilveeraghanta — COMMENTED** [review]
nice work! make sure to fill out the PR template with the resolves #yourticketnumberhere

**Review by jonathanlew — COMMENTED** [review]
nothing to add, looks good

**Review by jonathanlew — APPROVED** [review]

**Pull request merged**